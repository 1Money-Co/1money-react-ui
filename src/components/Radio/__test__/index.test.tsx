import 'jsdom-global/register';
import * as React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import { Radio } from '../index';

const originalConsoleError = console.error;
console.error = (message, ...optionalParams) => {
  if (
    message.includes('Could not parse CSS stylesheet') ||
    message.includes('findDOMNode is deprecated and will be removed') ||
    message.includes('React does not recognize')
  ) {
      return;
  }
  originalConsoleError(message, ...optionalParams);
};

describe('Radio', () => {
  it('renders correctly', () => {
    const radio = render(
      <Radio id="test-radio" value="test" label="Test Radio" />
    );

    expect(radio).toMatchSnapshot();
    expect(screen.getByText('Test Radio')).toBeInTheDocument();
    expect(screen.getByRole('radio')).not.toBeChecked();
  });

  it('renders checked state correctly', () => {
    render(
      <Radio id="test-radio" value="test" label="Test Radio" checked />
    );

    expect(screen.getByRole('radio')).toBeChecked();
  });

  it('renders disabled state correctly', () => {
    render(
      <Radio id="test-radio" value="test" label="Test Radio" disabled />
    );

    expect(screen.getByRole('radio')).toBeDisabled();
  });

  it('renders invalid state correctly', () => {
    const { container } = render(
      <Radio id="test-radio" value="test" label="Test Radio" invalid />
    );

    // PrimeReact RadioButton会添加p-invalid类名
    const radioElement = container.querySelector('.p-invalid');
    expect(radioElement).toBeInTheDocument();
  });

  it('renders different sizes correctly', () => {
    const { rerender } = render(
      <Radio id="test-radio" value="test" label="Small Radio" size="sm" />
    );

    expect(screen.getByText('Small Radio')).toBeInTheDocument();
    expect(document.querySelector('.rd-sm')).toBeInTheDocument();

    rerender(
      <Radio id="test-radio" value="test" label="Medium Radio" size="md" />
    );
    expect(screen.getByText('Medium Radio')).toBeInTheDocument();
    expect(document.querySelector('.rd-md')).toBeInTheDocument();

    rerender(
      <Radio id="test-radio" value="test" label="Large Radio" size="lg" />
    );
    expect(screen.getByText('Large Radio')).toBeInTheDocument();
    expect(document.querySelector('.rd-lg')).toBeInTheDocument();
  });

  it('calls onChange when clicked', async () => {
    const handleChange = jest.fn();
    render(
      <Radio
        id="test-radio"
        value="test"
        label="Test Radio"
        onChange={handleChange}
      />
    );

    const user = userEvent.setup();
    const radio = screen.getByRole('radio');

    await user.click(radio);

    expect(handleChange).toHaveBeenCalled();
  });

  it('does not call onChange when disabled', async () => {
    const handleChange = jest.fn();
    render(
      <Radio
        id="test-radio"
        value="test"
        label="Test Radio"
        onChange={handleChange}
        disabled
      />
    );

    const user = userEvent.setup();
    const radio = screen.getByRole('radio');

    await user.click(radio);

    expect(handleChange).not.toHaveBeenCalled();
  });
});
