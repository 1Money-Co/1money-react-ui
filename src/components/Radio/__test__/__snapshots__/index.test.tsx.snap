// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="om-react-ui-radio-wrapper "
      >
        <div
          class="om-react-ui-radio-inner  rd-md"
        >
          <div
            class="om-react-ui-radio-radio p-radiobutton p-component"
            data-pc-name="radiobutton"
            data-pc-section="root"
          >
            <input
              class="p-radiobutton-input"
              data-pc-section="input"
              id="test-radio"
              type="radio"
            />
            <div
              class="p-radiobutton-box"
              data-pc-section="box"
            >
              <div
                class="p-radiobutton-icon"
                data-pc-section="icon"
              />
            </div>
          </div>
          <label
            class="om-react-ui-radio-label"
            for="test-radio"
          >
            Test Radio
          </label>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="om-react-ui-radio-wrapper "
    >
      <div
        class="om-react-ui-radio-inner  rd-md"
      >
        <div
          class="om-react-ui-radio-radio p-radiobutton p-component"
          data-pc-name="radiobutton"
          data-pc-section="root"
        >
          <input
            class="p-radiobutton-input"
            data-pc-section="input"
            id="test-radio"
            type="radio"
          />
          <div
            class="p-radiobutton-box"
            data-pc-section="box"
          >
            <div
              class="p-radiobutton-icon"
              data-pc-section="icon"
            />
          </div>
        </div>
        <label
          class="om-react-ui-radio-label"
          for="test-radio"
        >
          Test Radio
        </label>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
